addons/sale/models/sale_order_line.py

```diff
@@ -221,6 +221,11 @@ def write(self, values):
             precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
             self.filtered(
                 lambda r: r.state == 'sale' and float_compare(r.product_uom_qty, values['product_uom_qty'], precision_digits=precision) != 0)._update_line_quantity(values)
+            # Prevent computing the package type when we are setting the package quantity
+            if 'product_packaging_qty' in values and 'product_packaging_id' not in values:
+                self.env.remove_to_compute(self._fields['product_packaging_id'], self)
```

Comment:
Is this remove_to_compute necessary ? (the test doesn't fail when i remove this line)
AFAIK, the orm will only mark this field to be recomputed during the super call to `write`, at this point it won't be marked to recompute :thinking:

Reply:
You are right, I've just checked, and indeed, at that point in time, in `self.env.all.tocompute` the key `product_packaging_id` is not present, so removing it is a redundant step. 
I've removed the that line, since it doesn't have any effect.